src/lib.rs

```diff
@@ -200,11 +203,7 @@ impl LightningNode {
             let funding_outpoint = channel_monitor.get_funding_txo().0;
             match chain_monitor.watch_channel(funding_outpoint, channel_monitor) {
                 ChannelMonitorUpdateStatus::Completed => {}
-                ChannelMonitorUpdateStatus::InProgress => {
-                    return Err(permanent_failure(
-                        "Failed to give a ChannelMonitor to the ChainMonitor",
-                    ))
-                }
+                ChannelMonitorUpdateStatus::InProgress => {}
```

Comment:
Stopped returning an error as the `InProgress` status is not a problem. When using remote persistence, it's expected that the status is usually `InProgress` as a background task is launched to handle remote persistence.